operator/pkg/ciliumendpointbatch/reconciler.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2021 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ciliumendpointbatch
+
+import (
+	"context"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	cilium_v2 "github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2alpha1"
+	clientset "github.com/cilium/cilium/pkg/k8s/client/clientset/versioned/typed/cilium.io/v2alpha1"
+	"github.com/sirupsen/logrus"
+)
+
+// reconciler is used to sync the current (i.e. desired) state of the CEBs in datastore into current state CEBs in the k8s-apiserver.
+// The source of truth is in local datastore, client reconcile with apiserver at regular intervals.
+type reconciler struct {
+	client     clientset.CiliumV2alpha1Interface
+	cebManager cebManager
+}
+
+// newReconciler creates and initializes a new reconciler.
+func newReconciler(client clientset.CiliumV2alpha1Interface, cebMgr cebManager) *reconciler {
+	return &reconciler{
+		client:     client,
+		cebManager: cebMgr,
+	}
+}
+
+// reconcileWithServer gets the list of CEBs to create, update and delete and syncs with the api-server.
+// On error, it returns the list of failed createCebs, updateCebs and deleteCebs to the caller.
+func (r *reconciler) reconcileWithServer(cebToCreate, cebToUpdate, cebToDelete string) error {
```

Comment:
This function should be broken into 3:
- reconcileCebCreate
- reconcileCebUpdate
- reconcileCebDelete

There's nothing shared among them, mixing them together makes this function less ideal for readability and code maintenance.

Reply:
I prefer a simple/single function that reconciles based on type of argument.

Reply:
Ok, Done.